src/main/java/io/vavr/jackson/datatype/deserialize/OptionDeserializer.java

```diff
@@ -94,4 +113,32 @@ public void resolve(DeserializationContext ctxt) throws JsonMappingException {
     public Option<?> getNullValue(DeserializationContext ctxt) throws JsonMappingException {
         return Option.none();
     }
+
+    @Override
+    public JsonDeserializer<?> createContextual(DeserializationContext ctxt, BeanProperty property) throws JsonMappingException {
+        JsonDeserializer<?> deser = valueDeserializer;
+        TypeDeserializer typeDeser = valueTypeDeserializer;
+        JavaType refType = valueType;
+
+        if (deser == null) {
+            deser = ctxt.findContextualValueDeserializer(refType, property);
+        } else { // otherwise directly assigned, probably not contextual yet:
+            deser = ctxt.handleSecondaryContextualization(deser, property, refType);
+        }
+        if (typeDeser != null) {
+            typeDeser = typeDeser.forProperty(property);
+        }
+        return withResolved(refType, typeDeser, deser);
+    }
+
+    /**
+     * Overridable fluent factory method used for creating contextual
+     * instances.
+     */
+    private OptionDeserializer withResolved(JavaType refType, TypeDeserializer typeDeser, JsonDeserializer<?> valueDeser) {
+        if (refType == valueType && valueDeser == valueDeserializer && typeDeser == valueTypeDeserializer) {
+            return this;
+        }
+        return new OptionDeserializer(this, typeDeser, valueDeser);
```

Comment:
When doing this, we will lose the deserializers found in parent class attribute `ValueDeserializer#deserializers`. But I think it's fine because they are no longer used in this class: all the usage of this attribute are removed, e.g. `deserializer(0)`.